csh_ldap/__init__.py

```diff
@@ -26,10 +57,15 @@ def __init__(self, bind_dn, bind_pw, batch_mods=False,
                   "#                                      #\n"
                   "########################################")
         ldap_srvs = srvlookup.lookup("ldap", "tcp", self.__domain__)
-        ldap_uris = ""
-        for uri in ldap_srvs:
-            ldap_uris += "ldaps://"+uri.hostname+","
-        self.__con__ = ReconnectLDAPObject(ldap_uris)
+        self.ldap_uris = ''.join([f'ldaps://{uri.hostname},' for uri in ldap_srvs])
+        self.server_uri = None
+
+        for uri in self.ldap_uris.split(',')[:-1]:
```

Comment:
Why are you generating a list, joining it, then splitting it?
```suggestion
        self.ldap_uris = [f'ldaps://{uri.hostname}' for uri in ldap_srvs]
        self.server_uri = None

        for uri in self.ldap_uris:
```